### ISSUE TITLE
[Feature Request]: Move inis Folder to AppRoot

### ISSUE THREAD

**Issue by silentq** [PCSX2/pcsx2#11376]
### Description

Change the default location of the inis folder to the the AppRoot folder.

### Reason

Currently on a fresh install without portable.ini present the Application places the inis folder in the Documents folder.  While I understand the the intent in practice it can be a little problematic. Certain settings like global resolution or device or the games folder path will carry over to the next device which may not be correct for that device.  This can be messy as changing those settings on one device changes it on both and can be avoided by having the inis folder local to each device and all the other folders like cache, gamesettings, etc still live in the Documents folder for a better Cloud-like experience.

### Examples

This change could simply be accomplished by changing the following code in the Pcsx2Config.cpp

From:
![image](https://github.com/PCSX2/pcsx2/assets/2636426/3dfa10e0-1d16-443a-8a31-f39d24f23a8c)

To:
![inisfix](https://github.com/PCSX2/pcsx2/assets/2636426/7efe4d5d-25b3-4461-8dfd-5f21b45925da)

While this will only change on fresh install from my testing it successfully sets the inis folder to be in the AppRoot(App Folder) while still allowing all the other folders to be placed in the DataRoot(Documents.)

I don't know if it would be best practice to go any further than this on existing installs but I do know this accomplishes fixing the issue I mentioned above.

**Comment by stenzek** [comment]
No, because then PCSX2 would not be able to save it's settings if you ran it from a non writable location, such as program files.

**Comment by silentq** [comment]
> No, because then PCSX2 would not be able to save it's settings if you ran it from a non writable location, such as program files.

Ah I see what you mean.  Just out of curiosity would it be possible to give the inis a unique name per device then or would that confuse each device as what the correct inis folder is to use?

**Comment by TellowKrinkle** [comment]
Since INIs are hardcoded to the data dir, why not redirect everything else?

**Comment by stenzek** [comment]
For the use case of "different devices with different game directories", I don't see the issue with just adding all devices' paths to the game list. It's not like the path _must_ exist, the scanner will skip over it if it doesn't.

**Comment by silentq** [comment]
> Since INIs are hardcoded to the data dir, why not redirect everything else?  Is there a non-INIs folder you want to redirect but can't?

The other folders are not the problem I have no issue with them being shared in the documents folder since their effects won't hurt functionality on another device like global resolution or even games directory that kind of thing.

So my thinking originally was since the inis folder can't be changed within the settings that keeping separate per device would in practice make it less unruly.

@stenzek makes a great point the inis cannot be in a folder that doesn't have write access.

However my second suggestion above is initializing the inis folder with a unique name like inis0, inis1 etc. That way they are kept separate per device.

Maybe I am misunderstanding how this is intended to work, but am curious to know if it's feasible.

**Comment by stenzek** [comment]
It's not feasible; how would each device know which folder to use?

**Comment by RedDevilus** [comment]
So if i'm understanding you want a tailored set where you can pick configs and bind them to correct device. Which sounds like unnecessary extra steps personally, you can just copy or start from scratch and adjust. The default settings are pretty universal all across the board from platform to specs.

Unless you want to go the route that Steam does, which has Cloud saves for stuff like savegames and it can say if you want local/cloud and if it is out of sync.

Or more to roll it out like a central server and you can point what you want. I don't really understand how it would even pick the right configs for devices even if you want big differences in config.

Can you give some other examples like what specific settings one device and the other settings on another device? Not like it's like 1.6 days where you need to constantly change settings on a global scale either.

**Comment by silentq** [comment]
So as far as telling which device made the inis file you could go with a PC Name as a unique identifier.  Windows PC generally have a Device Name.

I think the functionality I am perhaps poorly trying to describe is more akin to how some games separate their save data from their settings data and only the save data is in the cloud.  The concept is not all devices may want their settings to be at 4K resolution or maybe the controls scheme on each device is different etc.

In PCSX2 case I envision for example some users maybe using a lower power device that has 1080p as its global resolution whereas high end rig may have it set to 4K.  Controls may be different.  Some users may not feel comfortable with bios in the cloud and have it stored locally on the machine.  Those kind of things are set on the ini file.  This can be a pain to have to constantly switch if not using a portable mode.

At the end of the day it's more of a QOL feature but I totally understand if it's not something worth integrating.

**Comment by RedDevilus** [comment]
Yeah, atleast you can easily switch to portable mode with the portable.ini (even txt works to make it foolproof for regular users)
[portable.zip](https://github.com/user-attachments/files/15782317/portable.zip)

To be frank, would be a QoL to put it in the wizard or GUI to also be able to auto-create or pick path if they want standard installation or portable installation more akin to 1.6 installer also had both options but made more accessible:

![image](https://github.com/PCSX2/pcsx2/assets/24227051/e02f701d-8731-4b37-a82b-2dcf610ac4ec)

Because not everyone will know the portable mode exists, nor the Commandline arguments. Maybe it's a different viewpoint but might be a similar vision you have. For your original request, it will depend if some version of PCSX2 gets it implemented in future.

**Comment by stenzek** [comment]
> So as far as telling which device made the inis file you could go with a PC Name as a unique identifier. Windows PC generally have a Device Name.

And what happens if you change your computer name, or reinstall Windows? Your config no longer loads. I would not expect changing a computer name to break an application's configuration. This is getting **way** too niche.

> In PCSX2 case I envision for example some users maybe using a lower power device that has 1080p as its global resolution whereas high end rig may have it set to 4K. Controls may be different. Some users may not feel comfortable with bios in the cloud and have it stored locally on the machine. 

Then use portable mode, and point the save etc directories at the synced directory.

**Comment by silentq** [comment]
@stenzek First off apologizes if I am reaching here a bit honestly just more curious than anything about all this.

You make a great point regarding when the computer name changes.  Honestly I was probably overthinking it.  I figured just for finality I would experiment with one more thing here revisiting my initial idea.  You had said inis was not put in the AppRoot due to write protections and I totally get it but what if you placed it elsewhere.

For example I notice a lot other applications will put their settings in something like AppData\Roaming.  Placing it here would ensure the inis folder is local to each device while not putting it in Documents\PCSX2 which is a folder that is by default in the Cloud for most.  AppData\LocalLow is another option however I believe this can be problematic since its per user and the folder can have issues on locked down PC's at least from my research.

I experimented a little on my end to see if this can be done I just wanted to share what I did to achieve it. 
All of this is in the Pcsx2Config.cpp:
![image](https://github.com/PCSX2/pcsx2/assets/2636426/32203aeb-8aaa-444d-9dd5-da2063c1329b)
![image](https://github.com/PCSX2/pcsx2/assets/2636426/7bccd7f8-e9cc-4a56-828b-cfa07605f8ef)
![image](https://github.com/PCSX2/pcsx2/assets/2636426/888a30cb-56ab-4fbb-93ca-43647841eddd)

Sorry again for pushing a bit here I just wanted to leave this all here for posterity.  I completely understand if this never gets implemented.
